### PR TITLE
helix upgrade

### PR DESCRIPTION
**Pull request by marbelarrietaGlobant**
### Purpose

* updates to helix 2.24.0
* recompiles shaders
* updates hlsl common classes that our shaders reference 
* move visualization tests from /src to /test
* removes all net48 conditional code and references related to helix.

### Declarations

Check these if you believe they are true

- [ ] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated
- [ ] This PR contains no files larger than 50 MB 

### Release Notes

(FILL ME IN) Brief description of the fix / enhancement. **Mandatory section** 

### Reviewers

(FILL ME IN) Reviewer 1  (If possible, assign the Reviewer for the PR)

(FILL ME IN, optional) Any additional notes to reviewers or testers.

### FYIs

(FILL ME IN, Optional) Names of anyone else you wish to be notified of

### CODE REVIEW

**Comment by sm6srw** [comment]
I see this failure on the 4.8 build: E:\Builds\Dynamo_master\Dynamo\src\DynamoCoreWpf\DynamoCoreWpf.csproj(1722,3): error MSB4025: The project file could not be loaded. The 'ItemGroup' start tag on line 134 position 4 does not match the end tag of 'Project'. Line 1722, position 3

**Comment by marbelarrietaGlobant** [comment]
> I see this failure on the 4.8 build: E:\Builds\Dynamo_master\Dynamo\src\DynamoCoreWpf\DynamoCoreWpf.csproj(1722,3): error MSB4025: The project file could not be loaded. The 'ItemGroup' start tag on line 134 position 4 does not match the end tag of 'Project'. Line 1722, position 3

@sm6srw  I can't replicate that, actually DynamoCoreWpf.csproj doesn't have a 1722 line. Can you please give me more details?

**Comment by sm6srw** [comment]
@marbelarrietaGlobant I have only seen it on one of the build machines. Can you resolve the conflicts and push again? Thanks!

File: src/DynamoCoreWpf/DynamoCoreWpf.csproj
Comment (pinzart90) [inline src/DynamoCoreWpf/DynamoCoreWpf.csproj:127]: please put the helixToolkit in a common itemGroup (to avoid this duplication)

File: src/DynamoCoreWpf/DynamoCoreWpf.csproj
Comment (pinzart90) [inline src/DynamoCoreWpf/DynamoCoreWpf.csproj:163]: Can these references be moved be under net48 ?

File: src/DynamoCoreWpf/DynamoCoreWpf.csproj
Comment (pinzart90) [inline src/DynamoCoreWpf/DynamoCoreWpf.csproj:119]: HelixToolkit.Core.Wpf and HelixToolkit.SharpDX.Core.Wpf are only targeting netcoreapp3.1
We can either make sure (as much as we can by building locally, testing etc) that these are compatible with net6 or we try to get a proper net6 nuget (contribute to the open source or fork it) 
@mjkkirschner @sm6srw @aparajit-pratap thoughts ?

Comment (mjkkirschner) [reply src/DynamoCoreWpf/DynamoCoreWpf.csproj:119]: I think the maintainers of helix are quite responsive - but I know the project is complex, considering the background geometry preview is a core component, it would be ideal if we could ensure it works - we don't have bandwidth or time to move to something else now - it seems the testing time could be spent working on the migration to net6 and if they don't accept it we can always use the fork locally until we can get it merged or replace with something else (usd, webgpu, etc etc) that is cross platform.

I think a spike ascertaining how difficult it is makes sense before making this decision.

Comment (mjkkirschner) [reply src/DynamoCoreWpf/DynamoCoreWpf.csproj:119]: perhaps in the meantime we can merge this and @avidit can manually evaluate the image comparison tests to make sure they are the same in net6 buids?

File: src/Libraries/Watch3DNodeModelsWpf/Watch3DNodeModelsWpf.csproj
Comment (mjkkirschner) [inline src/Libraries/Watch3DNodeModelsWpf/Watch3DNodeModelsWpf.csproj:43]: I think this needs to be removed / unified with @pinzart90 's latest changes.

File: src/DynamoCoreWpf/DynamoCoreWpf.csproj
Comment (mjkkirschner) [reply src/DynamoCoreWpf/DynamoCoreWpf.csproj:121]: @jasonstratton @pinzart90 @marbelarrietaGlobant ... seems we are running out of time to try these alternative approaches that are discussed here.

@marbelarrietaGlobant what is the state of this PR?

Comment (marbelarrietaGlobant) [reply src/DynamoCoreWpf/DynamoCoreWpf.csproj:121]: @mjkkirschner issues with the PR were solved. Just waiting to approve and merge or another revision.

File: src/DynamoCoreWpf/DynamoCoreWpf.csproj
Comment (aparajit-pratap) [inline src/DynamoCoreWpf/DynamoCoreWpf.csproj:121]: So are these replacements for `HelixToolkit.Wpf` and `HelixToolkit.Wpf.SharpDX` that are compatible with .net6?

File: src/DynamoCoreWpf/Rendering/HelixRenderPackage.cs
Comment (aparajit-pratap) [inline src/DynamoCoreWpf/Rendering/HelixRenderPackage.cs:6]: Is this namespace still used for .net6 or only net framework?

Comment (aparajit-pratap) [reply src/DynamoCoreWpf/Rendering/HelixRenderPackage.cs:6]: I believe this needs to be conditionally used only for .net framework.

File: src/DynamoCoreWpf/ViewModels/Watch3D/DynamoGeometryModel3D.cs
Comment (aparajit-pratap) [inline src/DynamoCoreWpf/ViewModels/Watch3D/DynamoGeometryModel3D.cs:100]: Is this function signature change valid only for .net6 or .net framework as well?

File: src/DynamoCoreWpf/ViewModels/Watch3D/DynamoPointLineRenderCore.cs
Comment (aparajit-pratap) [inline src/DynamoCoreWpf/ViewModels/Watch3D/DynamoPointLineRenderCore.cs:2]: This needs to be moved to the `#else` block too?

File: src/DynamoCoreWpf/ViewModels/Watch3D/HelixWatch3DViewModel.cs
Comment (aparajit-pratap) [inline src/DynamoCoreWpf/ViewModels/Watch3D/HelixWatch3DViewModel.cs:54]: This seems to be common to both.

File: src/Libraries/Watch3DNodeModelsWpf/Watch3DNodeModelsWpf.csproj
Comment (aparajit-pratap) [inline src/Libraries/Watch3DNodeModelsWpf/Watch3DNodeModelsWpf.csproj:30]: This is common.

File: src/VisualizationTests/WpfVisualizationTests.csproj
Comment (aparajit-pratap) [inline src/VisualizationTests/WpfVisualizationTests.csproj:20]: Common

File: src/DynamoCoreWpf/DynamoCoreWpf.csproj
Comment (marbelarrietaGlobant) [reply src/DynamoCoreWpf/DynamoCoreWpf.csproj:121]: I think so https://www.nuget.org/packages/HelixToolkit.SharpDX.Core.Wpf/

File: src/DynamoCoreWpf/ViewModels/Watch3D/DynamoGeometryModel3D.cs
Comment (marbelarrietaGlobant) [reply src/DynamoCoreWpf/ViewModels/Watch3D/DynamoGeometryModel3D.cs:100]: I updated HelixToolKit for .net framework too. In fact the update in .net 6 is almost the same but with different namespaces. If you go to the library repository you will notice that they change the namespace to .net and make some conditions. So, the change is valid for both frameworks

File: src/DynamoCoreWpf/ViewModels/Watch3D/DynamoPointLineRenderCore.cs
Comment (marbelarrietaGlobant) [reply src/DynamoCoreWpf/ViewModels/Watch3D/DynamoPointLineRenderCore.cs:2]: It is a common using that implements GeometryModel3D

**Review by sm6srw — APPROVED** [review]
LGTM!

File: src/DynamoCoreWpf/ViewModels/Watch3D/HelixWatch3DViewModel.cs
Comment (pinzart90) [inline src/DynamoCoreWpf/ViewModels/Watch3D/HelixWatch3DViewModel.cs:229]: so we have 32 bit support here just because this code has been copy pasted right >?

**Review by pinzart90 — APPROVED** [review]
one question about NVidia native binary import then LGTM

File: src/DynamoCoreWpf/ViewModels/Watch3D/HelixWatch3DViewModel.cs
Comment (mjkkirschner) [reply src/DynamoCoreWpf/ViewModels/Watch3D/HelixWatch3DViewModel.cs:229]: yeah, I also asked about this class being removed here:
https://github.com/helix-toolkit/helix-toolkit/issues/2004 ... so - not sure it will work.

you can give this a read if you are interested - though it seems we might be able to do something now that .net supports native external callers ...

anyway, that blog post and this method don't seem to work the same way so there must be more to it than an exported member in the pe file. I guess?

For the time being, I think this is safe enough even if it does not work - we'll just have to pay attention to reports of users needing to explicitly set hardware accel again for sandbox - like back in the dark ages 😉 )

**Pull request merged**
